services/src/unit-test/java/uk/gov/moj/sdt/services/messaging/MessageWriterTest.java

```diff
@@ -110,87 +111,73 @@ public void testQueueMessageWithEmptyTargetApp() {
      * to queue name map is not valid
      */
     @Test
-    public void testQueueMessageWithInvalidTarget() {
+    void testQueueMessageWithInvalidTarget() {
         // Setup finished, now tell the mock what to expect.
         final ISdtMessage sdtMessage = new SdtMessage();
         sdtMessage.setSdtRequestReference("Test");
 
-        messageSender.sendMessage("UnitTestQueue", sdtMessage);
-        EasyMock.expectLastCall();
-
-        // Get ready to call the mock.
-        replay(messageSender);
+        messageSender.sendMessage(UNIT_TEST_QUEUE, sdtMessage);
 
         // Send the message.
         try {
             messageWriter.queueMessage(sdtMessage, "UnitTest", false);
-            Assert.fail("Should have thrown an Illegal Argument exception here as the target app code is invalid");
+            fail("Should have thrown an Illegal Argument exception here as the target app code is invalid");
         } catch (final IllegalArgumentException e) {
-            Assert.assertTrue("Target application code does not have a mapped queue name", true);
+            assertTrue(true, "Target application code does not have a mapped queue name");
         }
     }
 
     /**
      * Test for a valid queue message on the MessageWriter.
      */
     @Test
-    public void testQueueMessage() {
+    void testQueueMessage() {
         // Setup finished, now tell the mock what to expect.
         final ISdtMessage sdtMessage = new SdtMessage();
         sdtMessage.setSdtRequestReference("Test");
 
-        messageSender.sendMessage("UnitTestQueue", sdtMessage);
-        EasyMock.expectLastCall();
-
-        // Get ready to call the mock.
-        replay(messageSender);
+        messageSender.sendMessage(UNIT_TEST_QUEUE, sdtMessage);
```

Comment:
This line should be setting up the expected behaviour on the mocked messageSender, not invoking it.  This needs to be wrapped in either the Mockito doNothing() or doAnswer() methods and the verify() line below changed to expect only a single invocation instead of 2.

Reply:
we don't need to call messageSender.sendMessage("UnitTestQueue.DLQ", sdtMessage);

removed this one now

and we can just do
verify(messageSender, times(1)).sendMessage(any(), any());

Reply:
Confirmed sendMessage() line removed and check made to verify that it has been invoked.